Dockerfile

```diff
@@ -0,0 +1,219 @@
+ARG PHP_VERSION=7.4
+ARG NODE_VERSION=12
+ARG NGINX_VERSION=1.19
+
+FROM php:${PHP_VERSION}-fpm-alpine AS sylius_php_base
+
+# persistent / runtime deps
+RUN apk add --no-cache \
+		acl \
+		file \
+		gettext \
+		git \
+		mariadb-client \
+		vim \
+	;
+
+ARG APCU_VERSION=5.1.17
+RUN set -eux; \
+	apk add --no-cache --virtual .build-deps \
+		$PHPIZE_DEPS \
+		coreutils \
+		freetype-dev \
+		icu-dev \
+		libjpeg-turbo-dev \
+		libpng-dev \
+		libtool \
+		libwebp-dev \
+		libzip-dev \
+		mariadb-dev \
+		zlib-dev \
+	; \
+	\
+	if [ ! -z "${PHP_VERSION}" ] && [ "${PHP_VERSION:0:3}" = "7.3" ]; then \
+		docker-php-ext-configure gd --with-jpeg-dir=/usr/include/ --with-png-dir=/usr/include --with-webp-dir=/usr/include --with-freetype-dir=/usr/include/; \
+		docker-php-ext-configure zip --with-libzip; \
+	else \
+		docker-php-ext-configure gd --with-jpeg --with-webp --with-freetype; \
+		docker-php-ext-configure zip --with-zip; \
+	fi; \
+	docker-php-ext-install -j$(nproc) \
+		exif \
+		gd \
+		intl \
+		pdo_mysql \
+		zip \
+	; \
+	pecl install \
+		apcu-${APCU_VERSION} \
+	; \
+	pecl clear-cache; \
+	docker-php-ext-enable \
+		apcu \
+		opcache \
+	; \
+	\
+	runDeps="$( \
+		scanelf --needed --nobanner --format '%n#p' --recursive /usr/local/lib/php/extensions \
+			| tr ',' '\n' \
+			| sort -u \
+			| awk 'system("[ -e /usr/local/lib/" $1 " ]") == 0 { next } { print "so:" $1 }' \
+	)"; \
+	apk add --no-cache --virtual .sylius-phpexts-rundeps $runDeps; \
+	\
+	apk del .build-deps
+
+ARG BLACKFIRE=false
+
+RUN set -eux; \
+	if [ ! -z "$BLACKFIRE" ] && [ "$BLACKFIRE" = "true" ]; then \
+		curl -A "Docker" -o /tmp/blackfire-probe.tar.gz -D - -L -s https://blackfire.io/api/v1/releases/probe/php/alpine/amd64/$(php -r "echo PHP_MAJOR_VERSION.PHP_MINOR_VERSION;"); \
+		mkdir -p /tmp/blackfire; \
+		tar zxpf /tmp/blackfire-probe.tar.gz -C /tmp/blackfire; \
+		mv /tmp/blackfire/blackfire-*.so $(php -r "echo ini_get ('extension_dir');")/blackfire.so; \
+		printf "extension=blackfire.so\nblackfire.agent_socket=tcp://blackfire:8707\n" > $PHP_INI_DIR/conf.d/blackfire.ini; \
+		rm -rf /tmp/blackfire /tmp/blackfire-probe.tar.gz; \
+		mkdir -p /tmp/blackfire; \
+		curl -A "Docker" -L https://blackfire.io/api/v1/releases/client/linux_static/amd64 | tar zxp -C /tmp/blackfire; \
+		mv /tmp/blackfire/blackfire /usr/bin/blackfire; \
+		rm -Rf /tmp/blackfire; \
+	fi
+
+COPY --from=composer:latest /usr/bin/composer /usr/bin/composer
+
+# https://getcomposer.org/doc/03-cli.md#composer-allow-superuser
+ENV COMPOSER_ALLOW_SUPERUSER=1
+RUN set -eux; \
+	composer global require "hirak/prestissimo:^0.3" --prefer-dist --no-progress --no-suggest --classmap-authoritative; \
+	composer clear-cache
+ENV PATH="${PATH}:/root/.composer/vendor/bin"
```

Comment:
It is impossible to use hirak/prestissimo with composer:2. I think we should use composer:1 instead of composer:latest (for Sylius 1.7)

Reply:
Indeed, this was written before composer version 2 became the new "latest". I will make it easier to configure by introducing a new argument and using the version tag instead of the "latest" tag.